components/search/src/filter.rs

```diff
@@ -0,0 +1,205 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{
+    error::Error, JSONEngineBase, JSONEngineRecord, JSONEngineUrl, JSONEngineUrls,
+    JSONSearchConfigurationRecords, RefinedSearchConfig, SearchEngineDefinition, SearchEngineUrl,
+    SearchEngineUrls, SearchUserEnvironment,
+};
+
+impl SearchEngineUrl {
+    pub(crate) fn from_engine_url(url: &JSONEngineUrl) -> Self {
```

Comment:
I'm not sure if this is the best way of handling these conversions. There appears to be multiple ways, and this seemed to make sense?